### PR TITLE
Offset piezo / focus lock performance improvements

### PR DESCRIPTION
**Pull request by barentine**
Addresses issue #313, maybe.

**Is this a bugfix or an enhancement?**
enhancement for the sake of bugfix

**Proposed changes:**
- use `request.Session.get` instead of `request.get` for socket keep-alives in `RLPIDFocusLockClient` 
- use `request.Session.get` instead of `request.get` for socket keep-alives in `OffsetPiezoClient` 
- connect focus lock to `frameWrangler.onFrameGroup` rather than `onFrame`
- make `frameWrangler` polling interval an attribute so we can change it if we like
- change `frameWrangler._polling_interval` in init_htsms_focus_lock
- move frameWrangler signal doc into docstring and clarify the part that I misunderstood earlier
- add minimal (typing) docstrings to `microscope` and `GUIInitTask` to aid linter/IDE shortcuts

**Checklist:**

- [ ] Tested with numpy=1.14
- [ ] Tested on python 2.7 and 3.6
- [ ] Tested with wx=3.x and wx=4.x [if UI code]
- [ ] Does the PR avoid variable renaming in existing code, whitespace changes, and other forms of tidying? [There is a place for code tidying, but it makes reviewing 
much simpler if this is kept separate from functional changes]

If an enhancement (or non-trivial bugfix):

- [x] Has this been discussed in advance (feature request, PR proposal, email, or direct conversation)?
- [ ] Does this change how users interact with the software? How will these changes be communicated?
- [x] Does this maintain backwards compatibility with old data?
- [ ] Does this change the required dependencies?
- [ ] Are there any other side effects of the change?

### CODE REVIEW

**Comment by barentine** [comment]
converted to draft while I re-tune the htsms focus lock - performance is nice, might even be able to add in the fastGraph stuff again

**Comment by barentine** [comment]
things are running very nicely, though the D was annoying enough I left it at the ZN PI servo for now. Should be all set @David-Baddeley

File: PYME/Acquire/frameWrangler.py [hunk 1]
```diff
@@ -54,6 +54,31 @@
...
+        Called once per new-frame appearing in our buffer; used to pass
+        frame data to e.g. spoolers. Note that while onFrame gets called once
+        per new frame, the new frames are only checked for once per polling
+        cycle, meaning that this event will be fired N-times for each 
+        `onFrameGroup` event.
+    onFrameGroup : dispatch.Signal
+        Called on each new frame group (once per polling interval) - use for 
+        updateing GUIs etc.
+    onStop : dispatch.Signal
+    _polling_interval : float
```
Comment (David-Baddeley) [inline PYME/Acquire/frameWrangler.py]: It's a bit odd having a private attribute with no way of setting documented (leading underscore implies that it should not be accessed from outside the class)

Comment (David-Baddeley) [reply PYME/Acquire/frameWrangler.py]: This is also a bit of a 'here be dragons' type thing - I wouldn't want to encourage people to change this if it were to be made a public attribute as the results might not be super predictable.

**Comment by David-Baddeley** [comment]
Should also really get this updated and into core docs (graphically illustrates frame timing)
[PYMEAcquireSchem.pdf](https://github.com/python-microscopy/python-microscopy/files/4910444/PYMEAcquireSchem.pdf)

File: PYME/Acquire/frameWrangler.py (hunk 1 above)
Comment (David-Baddeley) [reply PYME/Acquire/frameWrangler.py]: Best approach might be to leave as an undocumented (or documented with a comment but not in docstring) private attribute.

**Pull request merged**